### DISCUSSION TITLE
NFC insulin pen upload

### DISCUSSION THREAD

**Discussion by Richspiral** [j-kaltes/Juggluco#11] [Ideas]
Hi, are there any plans to introduce NFC insulin pen uploads so we can upload our insulin pen history via NFC?

And have this info pushed through libreview so our clinics can view it?

Thanks for all your hard work... I really appreciate the app!

Best wishes
Rich

**Comment by j-kaltes** [comment]
I don't have one and they are not reimbursed by health insurance in the Netherlands, so I can't make them work with Juggluco.

**Reply by yoshimo** [reply]
you could check out https://github.com/NightscoutFoundation/xDrip/commit/a7bdacd622684d1fbfdfbaf1bf4554c552029e65 they support these pens for a while now

**Reply by JulGer** [reply]
Or just leave it as it is! 😉

**Comment by j-kaltes** [comment]
I made a version of Juggluco that can read via NFC NovoPen® 6 & NovoPen Echo®Plus insulin pens. I have tested it on a pen containing only 6 values. Can someone tell if it also works when a pen contains the values of the past 3 months?

EDIT:
New version:
http://jkaltes.byethost16.com/Juggluco/download.html
https://www.juggluco.nl/Juggluco/download.html

**Reply by Richspiral** [reply]
Thanks for implementing this. I've tested it and unfortunately it's not working. It asks me the content of the pen - I select fast insulin - and then it reports 0 doses saved. The pen's nfc feature is working with other software ok. Happy to try any suggested solutions / carry out more testing. Thanks again for your efforts.

**Reply by j-kaltes** [reply]
From what date did you request doses?

**Reply by Richspiral** [reply]
[![Screenshot_20230410-234010.png](https://user-images.githubusercontent.com/108764983/231012619-74d0f034-74c0-40e9-bb10-9ffa04013e40.png)

I don't get the option to select a date. It asks what the content is (see screenshot), I select fast acting insulin, and then it reports 0 doses saved.

**Reply by Richspiral** [reply]
It's working now after the most recent update. Thanks so much for efforts! 👍🏼
